packages/react-instantsearch/src/components/SearchBox.js

```diff
@@ -233,9 +274,22 @@ class SearchBox extends Component {
             {...searchInputEvents}
             {...cx('input')}
           />
+          {this.props.showLoadingIndicator && (
+            <div
+              style={{
+                display: isSearchStalled ? 'block' : 'none',
+              }}
+              {...cx('loading-indicator')}
+            >
+              {loadingIndicatorComponent}
+            </div>
+          )}
```

Comment:
We could add a test for this logic. Verify depending on the given props that the loading component is displayed or not (with a Snapshot).